sa/storage-authority.go

```diff
@@ -333,21 +333,26 @@ func (ssa *SQLStorageAuthority) CountCertificatesByNames(domains []string, earli
 	return ret, nil
 }
 
+type countByNamesHolder struct {
+	Serial       string
+	ReversedName string
+	NotBefore    time.Time
+}
+
 // countCertificatesByNames returns, for a single domain, the count of
 // certificates issued in the given time range for that domain and its
-// subdomains.
+// subdomains. If multiple certificates with the same FQDN are found
+// only the most recently issued one is counted.
 // The highest count this function can return is 10,000. If there are more
 // certificates than that matching one of the provided domain names, it will return
 // TooManyCertificatesError.
 func (ssa *SQLStorageAuthority) countCertificatesByName(domain string, earliest, latest time.Time) (int, error) {
 	var count int64
 	const max = 10000
-	var serials []struct {
-		Serial string
-	}
+	var names []countByNamesHolder
 	_, err := ssa.dbMap.Select(
```

Comment:
I was under the impression we wanted to leave this rate limit as is but to add a new one that was for multiple certificates that would get a different rate limit?